sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -422,7 +422,20 @@ export class BatchingReceiver extends MessageReceiver {
         });
         this._init(rcvrOptions)
           .then(() => {
-            this._receiver!.on(ReceiverEvents.receiverDrained, onReceiveDrain);
+            // there's a really small window here where the receiver can be closed
+            // just after .init() but before this.
+            if (this._receiver == null) {
+              const errorMessage = getReceiverClosedErrorMsg(
```

Comment:
Am assuming that if we didnt do this, the timer runs out and we resolve the promise returned by `receiveMessage()` with an empty array.

You will get the same result if the receiver was closed after the link was initialized and we are in the middle of receiving messages.

I dont see a considerable value add with this change. The program does not hang, it does get resolved when the timer runs out.

Reply:
Prior to this change you'd get an TypeError exception since the underlying `this._receiver` was undefined and we had a `this._receiver!.on()` call.

We can go with something less exception-y :)

Reply:
Then I would follow the same approach as the streaming receiver and gracefully close instead of throwing an error 
- See https://github.com/Azure/azure-sdk-for-js/blob/741cb6361bc4390f6abd75f056bab12f4ec124ab/sdk/servicebus/service-bus/src/receiver.ts#L137-L140
- https://github.com/Azure/azure-sdk-for-js/blob/741cb6361bc4390f6abd75f056bab12f4ec124ab/sdk/servicebus/service-bus/src/core/messageReceiver.ts#L959-L967

In this case it would be to have the promise resolved

Reply:
I put in a small workaround here - ignore for a moment, I want to trace it a bit more.

Reply:
(the rest of the outer locking logic is all fair game at the moment, however)

Reply:
Okay, looking at this now - it looks okay. Basically if there's no receiver I just resolve to an empty array and exit. No credits were added on the receiver so I think we're okay with cleanup as well.